### PR TITLE
Test all available openssh cryptographic policies

### PR DESCRIPTION
**Pull request by ggkioulis**
- Related ticket: https://progress.opensuse.org/issues/80716
- Needles: No Needles
- Verification run: x86_64: [sle-15-SP3](http://angmar.suse.de/tests/3042#step/sshd/1) | [sle-15-SP2](http://angmar.suse.de/tests/3034#step/sshd/375) | [sle-15-SP1](http://angmar.suse.de/tests/3035#step/sshd/374) | [sle-15](http://angmar.suse.de/tests/3036#step/sshd/374) | [sle-12-SP5](http://angmar.suse.de/tests/3037#step/sshd/394) | [sle-12-SP4](http://angmar.suse.de/tests/3038#step/sshd/394) | [sle-12-SP3](http://angmar.suse.de/tests/3039#step/sshd/394) | [sle-12-SP2](http://angmar.suse.de/tests/3040/sshd/1) | [tumbleweed](http://angmar.suse.de/tests/3041#step/sshd/375)
s390x: [sle-15-SP2](https://openqa.suse.de/tests/5418959#step/sshd/375) | [sle-12-SP5](https://openqa.suse.de/tests/5420464/#step/sshd/1)

### CODE REVIEW

**Review by grisu48 — APPROVED** [review]
This is awesome man! :+1:

File: tests/console/sshd.pm [hunk 1]
```diff
@@ -136,6 +141,37 @@ sub run {
...
+    foreach my $policy (@policies) {
+        $policy->add_to_sshd_config();
+    }
+
+    my $sshd_config = script_output("cat /etc/ssh/sshd_config");
+    record_info("Restart sshd", $sshd_config);
+    systemctl("restart sshd");
+
+    # Add all the ssh public key hashes as known hosts
+    assert_script_run("ssh-keyscan -H localhost >> ~/.ssh/known_hosts");
```
Comment (pdostal) [inline tests/console/sshd.pm:167]: As there are only `localhost` keys in ~/.ssh/known_hosts` an you rescan all of them I'd do
````suggestion
    assert_script_run("ssh-keyscan -H localhost > ~/.ssh/known_hosts");
```

**Review by pdostal — APPROVED** [review]
You test all available ciphers, host key algorithms, key exchange algorithms and mac algorithms! This is amazing. Looking forward your VR!

**Comment by DimStar77** [comment]
Breaks all TW stagings as nmap is not part of the rings/stagings

https://openqa.opensuse.org/tests/overview?distri=opensuse&groupid=2&version=Staging%3AF&build=F.500.1

**Comment by ggkioulis** [comment]
> Breaks all TW stagings as nmap is not part of the rings/stagings
> 
> https://openqa.opensuse.org/tests/overview?distri=opensuse&groupid=2&version=Staging%3AF&build=F.500.1

Thanks for the heads up!

**Comment by DimStar77** [comment]
I am currently testing how much trouble we will end up in when adding nmap to ring1; so far it looks ok, so most likely we can just live with this - I'll report back soon

**Comment by pdostal** [comment]
> I am currently testing how much trouble we will end up in when adding nmap to ring1; so far it looks ok, so most likely we can just live with this - I'll report back soon

I will be opening a pull request today. I can fix it in the test. Maybe by:
```perl
`ypper_call 'in nmap';
```

**Comment by DimStar77** [comment]
is zypper_call 'in nmap' abstracted to work on Kubic/MicroOS/Transactional-Server?
We can't install on any such system (r/o FS) without reboots

**Comment by pdostal** [comment]
> is zypper_call 'in nmap' abstracted to work on Kubic/MicroOS/Transactional-Server?
> We can't install on any such system (r/o FS) without reboots

We already install `expect`.

**Comment by DimStar77** [comment]
actually, I think we have to veto this:


Request 867297 review
nmap >= 7.90 is nonfree

nmap will fall out of the standard distro and only be available as NonFree; we can't rely on non-free in Staging/Rings

**Pull request merged**